Sources/GRPC/ConnectionManager.swift

```diff
@@ -335,34 +379,21 @@ internal class ConnectionManager {
         guard case let .connecting(connecting) = self.state else {
           self.invalidState()
         }
-        channel = connecting.candidate
+        connecting.candidate
+          .whenComplete { _ in fulfillMuxPromiseGivenState(promise: muxPromise) }
```

Comment:
I'm not sure if this is correct: the channel future can complete before we receive channel active, i.e. we can still be in the connecting state when fulfilling the promise.

Reply:
This turned out to be quite a big change involving adding an extra promise to state.